components/merkledb/src/views/metadata.rs

```diff
@@ -318,7 +318,7 @@ where
         index_address: &IndexAddress,
         index_type: IndexType,
     ) -> Result<Self, Self> {
-        assert_valid_name(index_address.name());
+        assert_index_valid_full_name(index_address.name());
```

Comment:
I don't know whether this makes sense within this PR (IMO, it does), but it could make sense to refactor this constructor to return `Result<Self, AccessError>`, and add `AccessError` variants for an invalid name and for names used by the system (which could be defined as a name starting with `__`).